Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -1392,13 +1392,19 @@ void JitArm64::subfic(UGeckoInstruction inst)
   }
   else
   {
-    const bool allocate_reg = d == a;
-    gpr.BindToRegister(d, allocate_reg);
+    const bool will_read = d == a;
+    const bool is_zero = imm == 0;
+    const bool allocate_reg = will_read & !is_zero;
```

Comment:
Derp, this should be `&&`

Reply:
Oh yes, I missed that. Technically it will work anyway, but I recommend changing it.